projects/packages/jitm/src/js/jetpack-jitm.js

```diff
@@ -1,346 +1,372 @@
-
-				// Hide the JITM after 2 seconds.
-				setTimeout( function () {
-					$template.fadeOut( 'slow' );
-				}, 2000 );
-			} );
-		} );
-
-		// Handle CTA ajax actions.
-		$template.find( '.jitm-button[data-ajax-action]' ).on( 'click', function ( e ) {
-			e.preventDefault();
-			var button = $( this );
-			button.attr( 'disabled', true );
-			$.post( window.ajaxurl, {
-				action: button.data( 'ajax-action' ),
-				_nonce: $el.data( 'ajax-nonce' ),
-			} )
-				.done( function () {
-					$template.fadeOut( 'slow' );
-				} )
-				.fail( function () {
-					button.attr( 'disabled', false );
-				} );
-			return false;
-		} );
-
-		// Handle tracking for JITM CTA buttons
-		$template.find( '.jitm-button' ).on( 'click', function ( e ) {
-			var button = $( this );
-			var eventName = button.attr( 'data-jptracks-name' );
-			if ( undefined === eventName ) {
-				return;
-			}
-
-			var jitmName = button.attr( 'data-jptracks-prop' ) || false;
-			var messagePath = button.attr( 'data-jitm-path' ) || false;
-			var eventProp = {
-				clicked: jitmName,
-				jitm_message_path: messagePath,
-			};
-
-			if ( window.jpTracksAJAX ) {
-				window.jpTracksAJAX.record_ajax_event( eventName, 'click', eventProp );
-			}
-		} );
-	};
-
-	var reFetch = function () {
-		$( '.jetpack-jitm-message' ).each( function () {
-			var $el = $( this );
-
-			var message_path = $el.data( 'message-path' );
-			var query = $el.data( 'query' );
-			var redirect = $el.data( 'redirect' );
-			var hash = location.hash;
-
-			hash = hash.replace( /#\//, '_' );
-
-			// We always include the hash if this is My Jetpack page
-			if ( message_path.includes( 'jetpack_page_my-jetpack' ) ) {
-				message_path = message_path.replace(
-					'jetpack_page_my-jetpack',
-					'jetpack_page_my-jetpack' + hash
-				);
-			} else if ( '_dashboard' !== hash ) {
-				message_path = message_path.replace(
-					'toplevel_page_jetpack',
-					'toplevel_page_jetpack' + hash
-				);
-			}
-
-			var full_jp_logo_exists = $( '.jetpack-logo__masthead' ).length ? true : false;
-
-			apiFetch( {
-				path: addQueryArgs( JITM_ENDPOINT_URL, {
-					message_path,
-					query,
-					full_jp_logo_exists,
-				} ),
-				method: 'GET',
-			} ).then( function ( messages ) {
-				const message = messages?.[ 0 ];
-				if ( message?.content ) {
-					setJITMContent( $el, message, redirect );
-				}
-			} );
-		} );
-	};
-
-	reFetch();
-
-	$( window ).on( 'hashchange', function ( e ) {
-		const newURL = e.originalEvent.newURL;
-		const isJetpackPage = [
-			'jetpack',
-			'my-jetpack',
-			'jetpack-backup',
-			'jetpack-boost',
-			'jetpack-protect',
-			'jetpack-search',
-			'jetpack-social',
-			'jetpack-videopress',
-		].some( str => newURL.includes( `admin.php?page=${ str }` ) );
-
-		if ( isJetpackPage ) {
-			var jitm_card = document.querySelector( '.jitm-card' );
-			if ( jitm_card ) {
-				jitm_card.remove();
-			}
-			reFetch();
-		}
-	} );
-} );
+document.addEventListener('DOMContentLoaded', function() {
```

Comment:
You could use WP's `domReady`, which does [a couple of extra checks](https://github.com/WordPress/gutenberg/blob/7722d8e36f44a37839f908723115ad836280f486/packages/dom-ready/src/index.js#L36-L44):

```js
import domReady from '@wordpress/dom-ready';

domReady( function () {
```

Reply:
good call!